### PR TITLE
Enable nullable reference types on Avalonia.ReactiveUI project

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?
Enables nullable reference types on Avalonia.ReactiveUI project

## What is the current behavior?
There is no any nullable attributes on this project.

### CODE REVIEW

**Comment by maxkatz6** [comment]
@worldbeater can you take a look?

File: src/Avalonia.ReactiveUI/RoutedViewHost.cs [hunk 1]
```diff
@@ -67,12 +67,12 @@ public RoutedViewHost()
             {
                 var routerRemoved = this
                     .WhenAnyValue(x => x.Router)
-                    .Where(router => router == null)
-                    .Cast<object>();
+                    .Where(router => router == null)!
```
Comment (maxkatz6) [inline src/Avalonia.ReactiveUI/RoutedViewHost.cs:70]: Not sure why it doesn't allow to cast from Router? to object?.

Comment (worldbeater) [reply src/Avalonia.ReactiveUI/RoutedViewHost.cs:70]: Looks like yet another nullable reference inference bug that will be hopefully fixed in future .NET releases.

Worth noting that the `IObservable<T>` can never be null after a `Where` call.

**Review by worldbeater — APPROVED** [review]
Don't see any critical issues here, so 👍 and thanks for this.

**Review by grokys — APPROVED** [review]

**Pull request merged**